gateway/src/test/java/io/zeebe/gateway/api/job/LongPollingActivateJobsTest.java

```diff
@@ -176,11 +161,100 @@ public void shouldCompleteFollowingRequestsAfterTimeout() {
     verify(successRequest.getResponseObserver(), times(1)).onCompleted();
   }
 
+  @Test
+  public void shouldNotBlockOtherJobTypes() {
+    // given
+    final String otherType = "other-type";
+    stub.addAvailableJobs(otherType, 2);
+    final int threshold = 3;
+    activateJobsAndWaitUntilBlocked(threshold);
+
+    // when
+    final LongPollingActivateJobsRequest otherRequest =
+        getLongPollingActivateJobsRequest(otherType);
+    handler.activateJobs(otherRequest);
+
+    // then
+    verify(otherRequest.getResponseObserver(), timeout(1000).times(1)).onCompleted();
+  }
+
+  @Test
+  public void shouldProbeIfNoNotificationReceived() throws Exception {
+    // given
+    final long probeTimeout = 2000;
+    handler =
+        LongPollingActivateJobsHandler.newLongPollingActivateJobsHandler()
+            .setBrokerClient(brokerClient)
+            .setLongPollingTimeout(20000)
+            .setProbeTimeoutMillis(probeTimeout)
+            .build();
+    actorSchedulerRule.submitActor(handler);
+
+    final LongPollingActivateJobsRequest request = getLongPollingActivateJobsRequest();
+    handler.activateJobs(request);
+    waitUntil(() -> request.hasScheduledTimer());
+
+    // when
+    actorClock.addTime(Duration.ofMillis(probeTimeout));
+
+    // then
+    verify(stub, timeout(1000).times(2 * partitionsCount)).handle(any());
+  }
+
+  @Test
+  public void shouldProbeNextRequestWhenBlockedRequestsTimedOut() throws Exception {
+    // given
+    final long longPollingTimeout = 2000;
+    final long probeTimeout = 20000;
+    handler =
+        LongPollingActivateJobsHandler.newLongPollingActivateJobsHandler()
+            .setBrokerClient(brokerClient)
+            .setLongPollingTimeout(longPollingTimeout)
+            .setProbeTimeoutMillis(probeTimeout)
+            .build();
+    actorSchedulerRule.submitActor(handler);
+
+    final int threshold = 3;
+    final List<LongPollingActivateJobsRequest> requests =
+        activateJobsAndWaitUntilBlocked(threshold);
+
+    actorClock.addTime(Duration.ofMillis(longPollingTimeout));
+    requests.forEach(
+        request -> verify(request.getResponseObserver(), timeout(1000).times(1)).onCompleted());
+    Loggers.GATEWAY_LOGGER.info("Requests {}", requests);
+
+    // when
+    actorClock.addTime(Duration.ofMillis(probeTimeout));
+    Thread.sleep(100); // Give some time for the periodic probe to execute
```

Comment:
Is it here not possible to verify that the stub is invoked?

Reply:
No. Stub is not invoked because all blocked requests have timedout. We want to test here that, the next request is used for probe and not immediately blocked.